tests/EStimLibrary.UnitTests/Extensions/SpatialModel/StringHierarchy/StringHierarchyAreaFactoryTests.cs

```diff
@@ -0,0 +1,268 @@
+            new StringHierarchyAreaFactory(region);
+
+        // Create dictionary of parameter values
+        Dictionary<string, object> paramValues =
+            new Dictionary<string, object>();
+
+        // Create variable to store produced IArea
+        IArea product;
+
+        // Check TryCreate return and product
+        bool valid = factory.TryCreate(paramValues, out product, false);
+        Assert.False(valid);
+        Assert.Null(product);
+    }
+
+    /// <summary>
+    /// Test the TryCreate method with paramValues including a fullSpec value
+    /// and skipping value validation
+    /// </summary>
+    [Fact]
+    public void TryCreate_ShouldNotValidateShouldCreate()
+    {
+        // Create non-empty options, modifiers, and subregions parameters
+        var regionOptions = new HashSet<string>() { "right", "left" };
+        var regionModifiers = new Dictionary<string, HashSet<string>>
+        {
+            { "modSet1", new HashSet<string>() { "mod1", "mod2" } },
+            { "modSet2", new HashSet<string>() { "mod3", "mod4" } }
+        };
+        var regionSubregions = new Dictionary<string, StringHierarchyRegion>
+        {
+            { "child1", new StringHierarchyRegion("child1", null, null, null,
+                regionModifiers) },
+            { "child2", new StringHierarchyRegion("child2", null) }
+        };
+        // Create region with the above parameters
+        var region = new StringHierarchyRegion("base", null,
+            null, regionOptions, null, regionSubregions);
+
+        // Create factory with non-null base region
+        StringHierarchyAreaFactory factory =
+            new StringHierarchyAreaFactory(region);
+
+        // Create dictionary with valid spec
+        Dictionary<string, object> paramValues1 =
+            new Dictionary<string, object>(){
+                { "fullSpec", "left base, child1" } };
+
+        // Create dictionary with invalid spec
+        Dictionary<string, object> paramValues2 =
+            new Dictionary<string, object>() { { "fullSpec", "" } };
+
+        // Create variable to store produced IArea
+        IArea product;
+
+        // Check TryCreate return and product for valid spec
+        bool valid = factory.TryCreate(paramValues1, out product, true);
+        Assert.True(valid);
+        Assert.NotNull(product);
+        Assert.Equal((IArea) new StringHierarchyArea("left base, child1"),
+            product);
+
+        // Check TryCreate return and product for invalid spec
+        valid = factory.TryCreate(paramValues2, out product, true);
+        Assert.True(valid);
+        Assert.NotNull(product);
+        Assert.Equal((IArea) new StringHierarchyArea(""), product);
+    }
+
+    /// <summary>
+    /// Test the TryCreate method with paramValues including a fullSpec value
+    /// and not skipping value validation
+    /// </summary>
+    [Fact]
+    public void TryCreate_ShouldValidateShouldMaybeCreate()
```

Comment:
What is the "ShouldMaybe"? Are there some test values for which it should vs would not create? In that case, put them in separate test functions. This will be made easier by using [Theory] rather than [Fact] flags and then just having a bunch of different [InlineData()] test input options or all of them in one [MemberData()] flag. See some of the test classes merge into the hardware or release branch for examples. Restructuring to use these may or may not change which test methods you actually have, e.g., giving you one test method per condition tested, e.g., TryCreate_ValidatedValidParamsShouldCreate, TryCreate_UnvalidatedValidParamsShouldCreate, TryCreate_ValidatedInvalidParamsShouldFail. Then the different input values you give are different examples of each condition